File: apps/api/app/catalog/presentation/schemas/template_version_dto.py
"""Schema DTO d'une version de template (vue detail)."""

from datetime import date

from pydantic import BaseModel, Field


class TemplateVersionDTO(BaseModel):
    """Version provisionnable d'un template (exposee dans le detail)."""

    version: str = Field(..., description="Libelle de la version.")
    is_default: bool = Field(..., description="Version proposee par defaut.")
    is_lts: bool = Field(..., description="Version a support long terme (LTS).")
    eol_date: date | None = Field(None, description="Date de fin de vie (EOL), si connue.")

File: apps/web/src/catalog/types/dto/TemplateVersionDTO.ts
/**
 * Miroir d'une version de template dans `TemplateDetailDTO.versions[]`.
 * Contrat figé : `{version, is_default, is_lts, eol_date}`.
 */
export interface TemplateVersionDTO {
  version: string
  is_default: boolean
  is_lts: boolean
  /** Date de fin de support (ISO `YYYY-MM-DD`) ou `null` si inconnue. */
  eol_date: string | null
}
